scripts/check_compatibility.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+python_versions=("3.8" "3.9" "3.10" "3.11")
+all_passed=true
+
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+
+check_compatibility() {
+    python_version=$1
+    all_supported=0
+
+    while read -r requirement; do
+        package_name=$(echo "$requirement" | awk -F'[!=<>]' '{print $1}')
+        echo -n "Checking $package_name... "
+
+        url="https://pypi.org/pypi/$package_name/json"
+        response=$(curl -s $url)
+        status_code=$(curl -s -o /dev/null -w "%{http_code}" $url)
+
+        if [ "$status_code" != "200" ]; then
+            echo -e "${RED}Information not available${NC}"
+            continue
+        fi
+
+        classifiers=$(echo "$response" | jq -r '.info.classifiers[]')
+        requires_python=$(echo "$response" | jq -r '.info.requires_python')
+
+        base_version="Programming Language :: Python :: ${python_version%%.*}"
+        specific_version="Programming Language :: Python :: $python_version"
+
+        if echo "$classifiers" | grep -q "$specific_version" || echo "$classifiers" | grep -q "$base_version"; then
+            echo -e "${GREEN}Supported${NC}"
+        elif [ "$requires_python" != "null" ] && echo "$requires_python" | grep -Eq "==$python_version|>=$python_version|<=$python_version"; then
+            echo -e "${GREEN}Supported${NC}"
```

Comment:
do you think that knowing the reason of the failure could enrich the logs?? would it be valuable??

i mean, maybe 3.10 is not compatible bcz, we find a strict < 3.10, for example